nano/node/bootstrap/bootstrap.cpp

```diff
@@ -573,6 +592,159 @@ void nano::bootstrap_attempt::add_bulk_push_target (nano::block_hash const & hea
+			for (auto i = static_cast<CryptoPP::word32> (representatives.size () - 1); i > 0; --i)
+			{
+				auto k = nano::random_pool::generate_word32 (0, i);
+				std::swap (representatives[i], representatives[k]);
+			}
+			if (representatives.size () > reps_limit)
+			{
+				representatives.resize (reps_limit, nano::representative{ 0, 0, nullptr });
+			}
+		}
+		for (auto const & rep : representatives)
+		{
+			total_weight += rep.weight.number ();
+		}
+	}
+	// Select peers with total 25% of reps stake from top 50% of principal representatives
+	representatives_copy.resize (representatives_copy.size () / 2, nano::representative{ 0, 0, nullptr });
+	while (total_weight < reps_weight / 4) // 25%
+	{
+		auto k = nano::random_pool::generate_word32 (0, static_cast<CryptoPP::word32> (representatives_copy.size () - 1));
+		auto rep (representatives_copy[k]);
+		if (std::find (representatives.begin (), representatives.end (), rep) == representatives.end ())
+		{
+			representatives.push_back (rep);
+			total_weight += rep.weight.number ();
+		}
+	}
+	// Start requests
+	for (auto i (0), max_requests (20); i <= max_requests && !confirmed_frontiers && !stopped; ++i)
+	{
+		std::unordered_map<std::shared_ptr<nano::transport::channel>, std::deque<std::pair<nano::block_hash, nano::root>>> batched_confirm_req_bundle;
+		std::deque<std::pair<nano::block_hash, nano::root>> request;
+		// Find confirmed frontiers (tally > 12.5% of reps stake, 60% of requestsed reps responded
+		for (auto ii (frontiers.begin ()); ii != frontiers.end ();)
+		{
+			if (node->ledger.block_exists (*ii))
+			{
+				ii = frontiers.erase (ii);
+			}
+			else
+			{
+				nano::lock_guard<std::mutex> active_lock (node->active.mutex);
+				auto existing (node->active.find_inactive_votes_cache (*ii));
+				nano::uint128_t tally;
+				for (auto & voter : existing.voters)
+				{
+					tally += node->ledger.weight (voter);
+				}
+				if (existing.confirmed || (tally > reps_weight / 8 && existing.voters.size () >= representatives.size () * 0.6)) // 12.5% of weight, 60% of reps
+				{
+					ii = frontiers.erase (ii);
+				}
+				else
+				{
+					for (auto const & rep : representatives)
+					{
+						if (std::find (existing.voters.begin (), existing.voters.end (), rep.account) == existing.voters.end ())
+						{
+							release_assert (!(*ii).is_zero ());
+							auto rep_request (batched_confirm_req_bundle.find (rep.channel));
+							if (rep_request == batched_confirm_req_bundle.end ())
+							{
+								std::deque<std::pair<nano::block_hash, nano::root>> insert_root_hash = { std::make_pair (*ii, *ii) };
+								batched_confirm_req_bundle.emplace (rep.channel, insert_root_hash);
+							}
+							else
+							{
+								rep_request->second.emplace_back (*ii, *ii);
+							}
+						}
+					}
+					++ii;
+				}
+			}
+		}
+		if (frontiers.size () < frontiers_count * 0.1) // 90% of frontiers confirmed
```

Comment:
Maybe `0.1` can be defined somewhere. Suggest `required_frontier_confirmation_ratio = 0.9` and place it on the left size of condition)

Reply:
Changed. Is it more clear now?

Reply:
Perfect